### ISSUE TITLE
Include a check and warning to alert (or prevent) sending the wrong firmware over the air

### ISSUE THREAD

**Issue by georgevbsantiago** [thinger-io/vscode-extension#1]
Include a check and warning to alert (or prevent) when the Firmware sent to the Device via OTA has a “USERNAME”, “DEVICE_ID”, “DEVICE_CREDENTIAL” or “THINGER_SERVER” (and “APN_NAME”, “APN_USER”, “APN_PSWD”, and “SSID”, “SSID_PASSWORD”) different from the current Firmware that is in the device. This functionality is important to prevent (or alert) the Developer from updating a Firmware of Device “A” with the Firmware of a Device “B” that has a different configuration or identification (“USERNAME”, “DEVICE_ID”, “DEVICE_CREDENTIAL” or “ THINGER_SERVER", and “APN_NAME”, “APN_USER”, “APN_PSWD”, and “SSID”, “SSID_PASSWORD”). If this occurs, Device “A” may go offline and not be able to connect to the Thinger Server. This issue is even more critical when Device “A” is somewhere far away and difficult to access.
Ex: “USERNAME”, “DEVICE_ID”, “DEVICE_CREDENTIAL”, “THINGER_SERVER”, “APN_NAME”, “APN_USER”, “APN_PSWD”, “SSID”, “SSID_PASSWORD”

**Comment by georgevbsantiago** [comment]
Regarding this improvement in the OTA process, it might be interesting to consider two approaches.

First, define a directive/macro that registers the device model. Ex: Device A (#define THINGER_DEVICE_MODEL = 1 or #define THINGER_DEVICE_MODEL = "ULTRASOUND_PWM_SENSOR")

And a second directive/macro that defines whether the device can receive firmware from a different model. The default would be "TRUE" (#define THINGER_DEVICE_FIRMWARE_RECEPTION_TYPE = TRUE), meaning the device would accept the recording of different firmware models (Ex: The device has firmware for a level sensor. And it would accept firmware from a different model, a current sensor).

However, if it were set to FALSE (#define THINGER_DEVICE_FIRMWARE_RECEPTION_TYPE = FALSE), the device would only accept firmware from the same model that is currently programmed.

This implementation aims to prevent the user from mistakenly programming the firmware of Device A (level sensor) onto Device B (current sensor).

If there were a way to link the THINGER_DEVICE_FIRMWARE_RECEPTION_TYPE directive to other Thinger directives (e.g., “USERNAME”, “DEVICE_ID”, “DEVICE_CREDENTIAL”, or “THINGER_SERVER”), that would also be very useful, as these are complements that prevent Device A from receiving firmware with directives/macros different from those currently programmed into the firmware, which could cause the device to go offline if an improper change is made by mistake.

If the VS Code OTA extension identifies that the device does not allow the programming of another device model, it would display a popup, as is currently done when attempting to program a firmware with the same code version.

In short, the suggestion would be to create two safeguards.

The first would prevent the programming via OTA of firmware for Device A (e.g., "ultrasonic_PWM_level_sensor") with firmware for Device B (e.g., "current_sct013_sensor"). This would be a safeguard to prevent the device from malfunctioning due to firmware that is not compatible with its model.

The second would prevent the device from going offline due to changes in information that ensures its connection to the Thinger Server (e.g., "USERNAME", "DEVICE_ID", "DEVICE_CREDENTIAL", or "THINGER_SERVER").
